src/device.rs

```diff
@@ -227,8 +227,8 @@ impl Device {
         Ok(get_device_info(self.id(), DeviceInfo::CL_DEVICE_IMAGE3D_MAX_DEPTH)?.to_size())
     }
 
-    pub fn image_support(&self) -> Result<cl_uint> {
-        Ok(get_device_info(self.id(), DeviceInfo::CL_DEVICE_IMAGE_SUPPORT)?.to_uint())
+    pub fn image_support(&self) -> Result<bool> {
+        Ok(get_device_info(self.id(), DeviceInfo::CL_DEVICE_IMAGE_SUPPORT)?.to_uint() == CL_TRUE)
```

Comment:
Originally in `C`, where integers were used to represent booleans: zero was `false` and *anything* else was `true`.
Although the calls should return `CL_TRUE` for true, they can only be relied on to return `CL_FALSE` for false.
Therefore the conversion should be `value != CL_FALSE`.